src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/MessageBox.cs

```diff
@@ -1,695 +1,705 @@
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(
+            string messageBoxText, 
+            string caption, 
+            MessageBoxButton button, 
+            MessageBoxImage icon, 
+            MessageBoxResult defaultResult, 
+            MessageBoxOptions options) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, caption, button, icon, defaultResult, options);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with specified text, caption, and style.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(
+            string messageBoxText, 
+            string caption, 
+            MessageBoxButton button, 
+            MessageBoxImage icon, 
+            MessageBoxResult defaultResult) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, caption, button, icon, defaultResult, 0);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with specified text, caption, and style.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(
+            string messageBoxText, 
+            string caption, 
+            MessageBoxButton button, 
+            MessageBoxImage icon) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, caption, button, icon, 0, 0);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with specified text, caption, and style.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(
+            string messageBoxText, 
+            string caption, 
+            MessageBoxButton button) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, caption, button, MessageBoxImage.None, 0, 0);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with specified text and caption.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(string messageBoxText, string caption) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, caption, MessageBoxButton.OK, MessageBoxImage.None, 0, 0);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with specified text.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(string messageBoxText) 
+        {
+            return ShowCore(IntPtr.Zero, messageBoxText, String.Empty, MessageBoxButton.OK, MessageBoxImage.None, 0, 0);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Displays a message box with the string representation of the specified object.
+        ///    </para>
+        /// </devdoc>
+        public static MessageBoxResult Show(object messageBoxObject)

```

Comment:
Sorry, I do not think it is the good design. Because this override do as `public static MessageBoxResult Show(string messageBoxText)` can do, and this override hide the implementation details.